tests/PHPStan/Type/Php/data/sprintf.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace PHPStan\Type\Php\Test;
+
+use function PHPStan\Testing\assertType;
+
+$foo = 'foo';
+assertType("'foo'", sprintf('%s', $foo));
+
+function foo (string $nonLiteral): void {
+	assertType('non-empty-string', sprintf('%s', $nonLiteral));
+}
+
+/** @var literal-string $literal */
+$literal = '';
+assertType('literal-string&non-empty-string', sprintf('%s', $literal));
+
+function bar (int $int): void {
+	assertType('literal-string&non-empty-string', sprintf('%d', $int));
```

Comment:
Imo it can only return a literal string when the args to sprintf were literal.. but a regular int is not literal per se

Reply:
Thanks Markus; and for anyone wondering why (frustrating as it is), we discussed this in [Issue 6033](https://github.com/phpstan/phpstan/issues/6033).

Reply:
Yeah I think I got a bit carried away. I'll fix it, unless the rest of the discussion leads to this PR being closed.